internal/biz/resources/resources.go

```diff
@@ -281,34 +282,56 @@ func (uc *Usecase) CheckForUpdate(ctx context.Context, permission, namespace str
 }
 
 func (uc *Usecase) ListResourcesInWorkspace(ctx context.Context, permission, namespace string, sub *kessel.SubjectReference, id string) (chan *model.Resource, chan error, error) {
-	resource_chan := make(chan *model.Resource)
-	error_chan := make(chan error, 1)
-
 	resources, err := uc.reporterResourceRepository.FindByWorkspaceId(ctx, id)
 	if err != nil {
 		return nil, nil, err
 	}
 
 	log.Infof("ListResourcesInWorkspace: resources %+v", resources)
 
+	const NUM_WORKERS = 10
+
+	resourceChan := make(chan *model.Resource, len(resources))
+	allowedChan := make(chan *model.Resource, len(resources))
+	errorChan := make(chan error, 1)
+	var wg sync.WaitGroup
+
+	// start workers
+	for i := 0; i < NUM_WORKERS; i++ {
+		go uc.checkWorker(ctx, permission, namespace, sub, resourceChan, allowedChan, errorChan, &wg)
+	}
+
+	wg.Add(len(resources))
```

Comment:
this should be"NUM_WORKERS" ?

Reply:
Since we are spanning only "n" number of workers and wait for them to finish

Reply:
We are waiting for all resources to be processed not for all workers to finish.

Reply:
clarified with Mark and we are good to go